solc/CommandLineInterface.cpp

```diff
@@ -835,6 +904,7 @@ void CommandLineInterface::compile()
 
 void CommandLineInterface::handleCombinedJSON()
 {
+	solAssert(m_assemblyStack);
```

Comment:
No, at least not at this spot. The function is supposed to work with and without `m_compiler` (it could have the EVM stack instead).

But it would be good to assert `m_compiler` in those `if`s that depend on it. That's a lot of repetition though - you'd have to refactor it first to extract the success condition into a separate `if`.